sphinx/source/governance.rst

```diff
@@ -94,3 +108,19 @@ At any stage during the voting process and before the proposal is completed, the
 
 This means future votes will be ignored, and the proposal will never be accepted. However it will remain visible as a proposal so members can easily audit historic proposals.
 
+Updating trusted enclave code versions
+--------------------------------------
+
+For new nodes to be able to join the network, the version of the code they run (as specified by the ``--enclave-file``) should be first trusted by the consortium of members.
+
+If the version of the code being executed needs to be updated (for example, to support additional endpoints), members can create a ``new_code`` proposal, specifying the new code version (e.g. ``3175971c02d00c1a8f9dd23ca89e64955c5caa94e24f4a3a0579dcfb2e6aebf9``):
+
+.. code-block:: bash
+
+    memberclient --cert member_cert --privk member_privk --rpc-address node_ip:node_port --ca network_cert add_code --new-code-id code_version
+
+Once the proposal has been accepted, nodes running the new code are authorised join the network. Nodes running older versions of the code can then be retired and stopped.
+
+.. note:: It is important to keep the code compatible with the previous version, since there will be a point in time in which the new code is running on at least one node, while the other version is running on a different node.
+
+.. note:: The safest way to restart or replace nodes is by stopping a single node running the old version and starting a node running the new version as a sequence of operations, in order to avoid a situation in which most nodes have been stopped, and new nodes will not be able to join since it would be impossible to reach a majority of nodes agreeing to accept new nodes (this restriction is imposed by the consensus algorithm).
```

Comment:
I think this is unclear. The key point is that when replacing all nodes, it is necessary to keep a voting consensus alive - so generally nodes should be restarted one-by-one, waiting until the replacement can vote before removing the next node. But also, isn't it safer to first _add_ the new node (potentially increasing our fault tolerance), rather than first _removing_ the old node (potentially decreasing our fault tolerance)?